backend/YouTubeV2.Application/Services/UserService.cs

```diff
@@ -53,5 +60,178 @@ public async Task RegisterAsync(RegisterDto registerDto, CancellationToken cance
             await _blobImageService.UploadProfilePictureAsync(image, user.Id, cancellationToken);
         }
+
+        public async Task DeleteAsync(string userID, CancellationToken cancellationToken)
+        {
+            await _userIDValidator.ValidateAndThrowAsync(userID, cancellationToken);
+
+            var user = await _userManager.FindByIdAsync(userID);
+            var userRoles = await _userManager.GetRolesAsync(user);
+
+            foreach (var role in userRoles)
+            {
+                var roleResult = await _userManager.RemoveFromRoleAsync(user, role);
+                if (!roleResult.Succeeded)
+                    throw new BadRequestException(roleResult.Errors.Select(error => new ErrorResponseDTO(error.Description)));
+            }
+
+            var deleteResult = await _userManager.DeleteAsync(user);
+            if (!deleteResult.Succeeded)
+                throw new BadRequestException(deleteResult.Errors.Select(error => new ErrorResponseDTO(error.Description)));
+
+            await _blobImageService.DeleteProfilePictureAsync(user.Id, cancellationToken);
+        }
+
+        public async Task<UserDTO> GetAsync(string userID, CancellationToken cancellationToken)
+        {
+            await _userIDValidator.ValidateAndThrowAsync(userID, cancellationToken);
+
+            var user = await _userManager.FindByIdAsync(userID);
+
+            return await GetDTOForUser(user);
+        }
+        private async Task<UserDTO> GetDTOForUser(User user)
+        {
+            var imageUri = _blobImageService.GetProfilePicture(user.Id);
+            var topRoleName = await GetTopRoleForUserAsync(user);
+
+            return new UserDTO(user.Id, user.Email, user.UserName, user.Name, user.Surname, user.AccountBalance,
+                topRoleName, imageUri.ToString(), user.SubscriptionsCount);
+        }
+        private async Task<string> GetTopRoleForUserAsync(User user)
+        {
+            var userRoles = await _userManager.GetRolesAsync(user);
+            string topRoleName = userRoles.Contains(Role.Creator, StringComparer.InvariantCultureIgnoreCase) 
+                ? Role.Creator : Role.Simple;
+
+            return topRoleName;
+        }
+
+        public async Task EditAsync(UserDTO userDTO, CancellationToken cancellationToken)
+        {
+            await _userIDValidator.ValidateAndThrowAsync(userDTO.id);
+            await _userDTOValidator.ValidateAndThrowAsync(userDTO);
+
+            var user = await _userManager.FindByIdAsync(userDTO.id);
+
+            if (!userDTO.avatarImage.IsNullOrEmpty())
+            {
+                await _blobImageService.DeleteProfilePictureAsync(user.Id, cancellationToken);
+                byte[] image = Convert.FromBase64String(userDTO.avatarImage);
+                await _blobImageService.UploadProfilePictureAsync(image, user.Id, cancellationToken);
+            }
+
+            await VerifyEmailUniqueAsync(user);
+            await VerifyNameUniqueAsync(user);
```

Comment:
i agree but i am not sure i know how to do it (i need to pass a user with provided id to lambdas in `.MustAsync()` and i dont know how.  If it can be done help would be appreciated.).